dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/main/java/org/apache/dolphinscheduler/plugin/task/http/HttpTask.java

```diff
@@ -289,8 +293,20 @@ protected void setHeaders(HttpUriRequest request, List<HttpProperty> httpPropert
      */
     protected CloseableHttpClient createHttpClient() {
         final RequestConfig requestConfig = requestConfig();
-        HttpClientBuilder httpClientBuilder;
-        httpClientBuilder = HttpClients.custom().setDefaultRequestConfig(requestConfig);
+        HttpClientBuilder httpClientBuilder = HttpClients.custom();
+        if (httpParameters.getEnableProxy()) {
+            CredentialsProvider provider = new BasicCredentialsProvider();
+            AuthScope authScope = null;
```

Comment:
Basic Authentication does not always need proxy, just username and password could be the least parameter